### PR TITLE
feat: add version match API

### PR DESCRIPTION
**Pull request by starsz**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues
#1165

### New feature or improvement
- Describe the details and related test reports.

Add an API to check the version of manger-api and apisix.

### CODE REVIEW

**Review by nic-chen — APPROVED** [review]

File: api/internal/handler/tool/tool.go [hunk 1]
```diff
@@ -17,28 +17,48 @@
...
 
+type nodes struct {
+	Hostname string `json:"hostname"`
+	Version  string `json:"version"`
+}
+
+type VersionMatchOutput struct {
+	Matched          bool    `json:"matched"`
+	DashboardVersion string  `json:"dashboard_version"`
+	NotMatchedNodes  []nodes `json:"not_matched_nodes"`
```
Comment (juzhiyuan) [inline api/internal/handler/tool/tool.go:50]: recommend using `mismatched_nodes`

File: api/internal/handler/tool/tool.go [hunk 2]
```diff
@@ -48,3 +68,44 @@ func (h *Handler) Version(_ droplet.Context) (interface{}, error) {
...
+
+	output.NotMatchedNodes = notMatchedNodes
+	if len(output.NotMatchedNodes) == 0 {
+		output.Matched = true
+	} else {
+		// TODO: move this to utils
+		return &data.SpecCodeResponse{StatusCode: http.StatusOK, Response: data.Response{
+			Data:    &output,
+			Code:    2000001,
+			Message: "The version of manager-api and apisix are not matched.",
```
Comment (juzhiyuan) [inline api/internal/handler/tool/tool.go:106]:
```suggestion
			Message: "The manager-api and apache apisix are mismatched.",
```

File: api/internal/handler/tool/tool.go (hunk 1 above)
Comment (starsz) [reply api/internal/handler/tool/tool.go:50]: Good idea.

**Comment by juzhiyuan** [comment]
need more members to review this PR.

File: api/internal/utils/consts/versionMap.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package consts
+
+/*
+	Key is dashboard version
+	Val is apisix version
+*/
+var VersionMap = map[string]string{
```
Comment (imjoey) [inline api/internal/utils/consts/versionMap.go:23]: Shall we use `const` to define the `VersionMap` here?

Comment (starsz) [reply api/internal/utils/consts/versionMap.go:23]: Sorry. We can't. Because of `const` type only for boolean, number, string.

Comment (imjoey) [reply api/internal/utils/consts/versionMap.go:23]: Wow, my bad. Sorry.

**Review by imjoey — APPROVED** [review]
/lgtm

**Comment by starsz** [comment]
> /lgtm

Thank you for your code review. 😄

**Pull request merged**